xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-templates/src/main/resources/templates/html_displayer/entitytreenode/edit.vm

```diff
@@ -0,0 +1,84 @@
+## ---------------------------------------------------------------------------
+## See the NOTICE file distributed with this work for additional
+## information regarding copyright ownership.
+##
+## This is free software; you can redistribute it and/or modify it
+## under the terms of the GNU Lesser General Public License as
+## published by the Free Software Foundation; either version 2.1 of
+## the License, or (at your option) any later version.
+##
+## This software is distributed in the hope that it will be useful,
+## but WITHOUT ANY WARRANTY; without even the implied warranty of
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+## Lesser General Public License for more details.
+##
+## You should have received a copy of the GNU Lesser General Public
+## License along with this software; if not, write to the Free
+## Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+## 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+## ---------------------------------------------------------------------------
+#if ($displayer.mode == 'edit')
```

Comment:
Is this really needed? None of the existing edit templates do this check, AFAICS. The edit template is supposed to be called only on edit mode.

Reply:
I actually took that from the default template file: https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-templates/src/main/resources/templates/html_displayer/default.vm which is generally called as fallback.

Reply:
Yes, ``default.vm`` is the fallback for when there is no **action** specific template. But in your case you have an **edit** specific template, ``edit.vm``, that is supposed to be called **only** on the edit action. Generally speaking, you don't need to check the action in an action specific template (i.e. check for ``edit`` in ``edit.vm``).

Reply:
Right makes sense indeed. Ok I'll remove this if.